core/dbt/context/common.py

```diff
@@ -476,19 +464,20 @@ def generate_model(model, config, manifest, source_config, provider):
                             source_config, provider)
     # operations (hooks) don't get a 'this'
     if model.resource_type != NodeType.Operation:
-        this = get_this_relation(context['adapter'], config, model_dict)
+        this = get_this_relation(context['adapter'], config, model)
         context['this'] = this
     # overwrite schema/database if we have them, and hooks + sql
+    # the hooks should come in as dicts, at least for the `run_hooks` macro
+    # TODO: do we have to preserve this as backwards a compatibility thing?
```

Comment:
@drewbanin We have to leave these as dicts in the context if we want to be backwards compatible because of the run_hooks macro, right?